src/main.js

```diff
@@ -91,17 +91,9 @@ export class AwsClient {
       input = url
     }
     const signer = new AwsV4Signer(Object.assign({ url: input }, init, this, init && init.aws))
-    const signed = Object.assign({}, init, await signer.sign())
+    const signed = Object.assign(signer.body instanceof ReadableStream ? { duplex: 'half' } : {}, init, await signer.sign())
     delete signed.aws
-    try {
-      return new Request(signed.url.toString(), signed)
-    } catch (e) {
-      if (e instanceof TypeError) {
-        // https://bugs.chromium.org/p/chromium/issues/detail?id=1360943
```

Comment:
As you can see from here – it's a Chrome workaround specifically.

You'd need to check that `{ duplex: 'half' }` works on all platforms (and that it's appropriate)